### ISSUE TITLE
Facet sorting issue

### ISSUE THREAD

**Issue by emhagman** [dgraph-io/dgraph#2083]
**Dgraph v1.0.2
Docker (Running with 2048MB) inside instance**

I have a dataset of "profiles" and "jobs" where people know each other and can be "prospects" for a job. I am trying to find all prospects for a job and then sort them based on the "score" that is on the "prospect" facet.

Example query
```
# Find prospects for job 580 for group 85 sorted by resolved score to a job, ensure they are in network
{
  var(func: eq(id, 580)) @cascade @ignorereflex {
    firstProspects as hasProspect {
      profileId: id
      ~knows @filter(eq(group, 85)) {
        firstDegreeProfileId: id
      }
    }
  }
  var(func: eq(id, 580)) @cascade @ignorereflex {
    secondProspects as hasProspect {
      profileId: id
      ~knows {
        secondDegreeProfileId: id
        ~knows @filter(eq(group, 85)) {
          firstDegreeProfileId: id
        }
      }
    }
  }
  result(func: eq(id, 580)) {
    jobId: id
    jobTitle: title@en
    prospects: hasProspect @filter(uid(firstProspects, secondProspects)) @facets(s as score, orderdesc: score) {
      profileId: id
      score: val(s)
    }
  }
}
```

The query above returns 14 nodes with `orderdesc` and only 9 nodes with `orderasc` but when I remove the ordering, it returns all 12,000+ profiles that I know to be prospects via my RDBS.

Am I just using GraphQL+ wrong or am I right in assuming that the sort above should work?

UPDATE:
```
 {
  result(func: eq(id, 580)) {
    jobId: id
    jobTitle: title@en
    prospects: hasProspect @facets(s as score, orderdesc: score) {
      profileId: id
      score: val(s)
    }
  }
}
```

The simpler version of this query without uid() does in fact sort correctly by the facet.

UPDATE 2: 
I am running into the same issues if I use a reverse lookup as well

```
 result(func: eq(id, 580)) {
    id
    title@en
    prospects: ~prospect @filter(uid(firstProspects, secondProspects)) @facets(s as score, orderdesc: score) {
     	id
      score: val(s)
    }
  }
```

UPDATE 3:

Even simpler, it just seems that sorting by facet does not work when using uid().
```
{
  var(func: eq(id, 580)) @cascade @ignorereflex {
    prospects as hasProspect
  }
  result(func: eq(id, 580)) {
    jobId: id
    jobTitle: title@en
    prospects: hasProspect @filter(uid(prospects)) @facets(s as score, orderdesc: score)  {
      id: id
      score: val(s)
    }
  }
}
```

UPDATE 4:

So I figured out a query in which I return the results I want but I am not sure I understand why the above queries do not work. Anyways, here is what worked:

```
# Find prospects for job 580 for group 85 sorted by resolved score to a job, ensure they are in network
{
  var(func: eq(id, 580)) @cascade @ignorereflex {
    firstProspects as hasProspect {
      profileId: id
      ~knows @filter(eq(group, 85)) {
        firstDegreeProfileId: id
      }
    }
  }
  var(func: eq(id, 580)) @cascade @ignorereflex {
    secondProspects as hasProspect {
      profileId: id
      ~knows {
        secondDegreeProfileId: id
        ~knows @filter(eq(group, 85)) {
          firstDegreeProfileId: id
        }
      }
    }
  }
  var(func: eq(id, 580)) {
    hasProspect @filter(uid(firstProspects, secondProspects)) @facets(s as score) {
      scoreVar as math(s)
    }
  }
  Result(func: uid(scoreVar), orderdesc: val(scoreVar)) {
    id
    score: val(scoreVar)
  }
}
```

**Comment by pawanrawal** [comment]
Hey @emhagman 

Thanks for the detailed report. That made it easy to reproduce the issue. I have pushed a fix to master at 99fdf7aa1d0299e442abbe7833882761bd941d64. It's going to be part of the `nightly` release.

**Comment by sboorlagadda** [comment]
Fixed in v1.0.3
